pkg/controller/endpointslicemirroring/utils.go

```diff
@@ -186,6 +187,17 @@ func objectRefPtrEqual(ref1, ref2 *corev1.ObjectReference) bool {
 	return true
 }
 
+// stringPtrEqual returns true if a set of string pointers have equivalent values.
+func stringPtrEqual(ptr1, ptr2 *string) bool {
```

Comment:
we have `stringPtrChanged` in endpointslices/utils.go, @robscott if the goal is to consolidate this code does make sense to have only one and negate the logic in one of the places?

Reply:
Good catch, it would be good to use that util function instead.

Reply:
That is a good point. I wasn't sure where the function should go. Is `endpointslices/utils.go` the best place and it can be called in the other controller as well?

Reply:
I added a function to utils/pointer just yesterday, LOL.

Ideally we can all just use pointer.StringEqual()

Reply:
needs to be re-vendored, so  "later"